### ISSUE TITLE
SourceLink fails on macOS Mojave with "PlatformNotSupportedException: osx.10.14-x64"

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/sourcelink#112]
### Repro
Create the sample in the README of this repo and `dotnet build`

### Error
```
/Users/nmcmaster/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63102-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: The "Microsoft.Build.Tasks.Git.LocateRepository" task failed unexpectedly. [/Users/nmcmaster/dev/natemcmaster/dotnet-serve/src/dotnet-serve/dotnet-serve.csproj]
/Users/nmcmaster/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63102-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.PlatformNotSupportedException: osx.10.14-x64 [/Users/nmcmaster/dev/natemcmaster/dotnet-serve/src/dotnet-serve/dotnet-serve.csproj]
/Users/nmcmaster/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63102-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.RuntimeIdMap.GetNativeLibraryDirectoryName() in /_/src/Microsoft.Build.Tasks.Git/RuntimeIdMap.cs:line 22 [/Users/nmcmaster/dev/natemcmaster/dotnet-serve/src/dotnet-serve/dotnet-serve.csproj]
```

### Details
Using: 1.0.0-beta-63102-01

**Comment by tmat** [comment]
The platform `osx.10.14-x64` is not listed in https://raw.githubusercontent.com/dotnet/corefx/master/pkg/Microsoft.NETCore.Platforms/runtime.json.

**Comment by bording** [comment]
This is another example of why the RID list has to be kept up to date!

**Comment by tmat** [comment]
I don't think that's the solution. We shouldn't need to update SourceLink every time Apple releases new OS. I think we need to parse out the OS and fallback to the highest version we got in `runtime.json`, if we don't have a precise match.

**Comment by bording** [comment]
My point is that this a larger problem than just SourceLink. Someone trying to use LibGit2Sharp at all on macOS 10.14 will run into the exact same problem. The native library resolution mechanism built into .NET Core depends on an accurate RID graph to work properly.

**Comment by tmat** [comment]
@bording Absolutely. All this RID resolution and library loading needs to be implemented in .NET Core.

**Comment by bording** [comment]
Thinking about it a bit more, I believe this error does indicate a slight divergence from .NET Core's behavior already. 

When .NET Core can't find an entry in the RID graph for the actual RID of the OS, it falls back to the RID of the machine that the .NET Core binaries were compiled on. For linux, that means `linux-x64`, which is generally not going to be helpful when you need distro-specific binaries.

On macOS, this fallback RID IIRC is `osx.10.12-x64`. So .NET Core would find that in the RID graph and pick the `osx` binaries, which hopefully will work on 10.14 as well.

**Comment by natemcmaster** [comment]
I agree with @bording's updated comment - it would be best to follow what corehost does. Fallback to a preset RID if you can't find the current one in the rid graph. For reference, https://github.com/dotnet/core-setup/blob/b64f7fffbd14a3517186b9a9d5cc001ab6e5bde6/src/corehost/common/pal.h#L53-L73

**Comment by tmat** [comment]
Well, I can't do exactly the thing as corehost as I'm not building a native library. The build on any OS should produce the same managed library (in fact, bit-for-bit since our builds are fully deterministic).

That said, we can do a similar thing - it's basically what I wanted to do when I said "parse out the OS" from the current RID and pick based on that.

**Comment by natemcmaster** [comment]
I never suggested you should build a native library. I'm only suggesting you follow the _behavior_ of the native host.
